pkg/controller/controller.go

```diff
@@ -13,3 +14,13 @@ func AddToManager(m manager.Manager, opts manager.Options, fnList ...func(manage
 	}
 	return nil
 }
+
+// AddToManagerWithFeatureGate adds all Controllers to the Manager along with a feature gate accessor
+func AddToManagerWithFeatureGate(m manager.Manager, opts manager.Options, featureGateAccessor featuregates.FeatureGateAccess, fnList ...func(manager.Manager, manager.Options, featuregates.FeatureGateAccess) error) error {
```

Comment:
I think on this occasion I would probably just extend the existing AddToManager rather thann introducing a new one, this isn't shared outside this repo is it?

Reply:
I've reverted this back to `AddToManagerWithFeatureGates` as we share it across controllers without the paused condition currently, eg the machine healthcheck & nodelink controllers.